### PR TITLE
chore(GitHubActions): Extract Dockerfile to repository and build it

### PR DESCRIPTION
**Pull request by Tiboris**
Move the Dockerfile meant as test container to repo
and build it using the path to the filename.
push built container to the github registry
when it is a PR and to the quay.io with latest
tag when it is run on main branch.
Do all this only when it is not fork.

Signed-off-by: Tibor Dudlák <tdudlak@redhat.com>

### CODE REVIEW

File: .github/workflows/gating.yaml [hunk 1]
```diff
@@ -16,33 +17,106 @@ jobs:
...
-              echo 'RUN dnf -y install python3-sssdconfig maven unzip python3-pip git python3-netifaces python3-devel krb5-devel gcc sssd-dbus dbus-devel glibc glib2-devel dbus-daemon python-devel openldap-devel python-ipalib'
-              echo 'RUN dnf clean all'
-              echo 'COPY ipa-tuura ipa-tuura'
-              echo 'WORKDIR /ipa-tuura'
-              echo 'RUN pip install -r src/install/requirements.txt'
-              echo 'RUN pip install dbus-python python-ldap'
-              echo 'RUN python3 src/ipa-tuura/manage.py test'
-          } > podmanfile
-          podman build --tag gatingtest -f ./podmanfile
+          docker pull ${{ env.IMAGE_REGISTRY_QUAY }}:${{ env.IMAGE_TAG }}
```
Comment (f-trivino) [inline .github/workflows/gating.yaml:121]: podman?

Comment (Tiboris) [reply .github/workflows/gating.yaml:121]: yes we can use it there i just automatically wrote docker `bad habit`?
this piece of code is entrypoint for Anuja to test the things

Comment (Tiboris) [reply .github/workflows/gating.yaml]: resolved by using podman i hope

**Comment by Tiboris** [comment]
@f-trivino or maybe you to give this a blessing? :)

**Comment by amore17** [comment]
> @amore17 could you please have a look?

For this which credentials we need to store in github variables.
I am getting failure in phase:  `Push to Repositories`

**Comment by Tiboris** [comment]
example form @amore17 pr https://github.com/amore17/ipa-tuura/actions/runs/5131928143/jobs/9232650412

**Review by f-trivino — APPROVED** [review]
@Tiboris thanks for your pull request, this is a much better way how to handle containerfiles and images.

**Pull request merged**